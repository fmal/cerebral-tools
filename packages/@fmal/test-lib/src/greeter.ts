export class Greeter {
  name: string;

  constructor(name: string) {
    this.name = name;
  }

  greet(): string {
    const result = `Hello, ${this.name}!`;
    console.log(result);
    return result;
  }
}
