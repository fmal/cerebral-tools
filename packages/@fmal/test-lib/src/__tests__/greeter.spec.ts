import { Greeter } from '../greeter';

describe('Greeter', () => {
  let greeter: Greeter;

  beforeEach(() => {
    greeter = new Greeter('World');
  });

  it('greets', () => {
    const actual = greeter.greet();

    expect(actual).toMatchSnapshot();
  });

  it('logs the greeting each time greet is called', () => {
    const spyLog = jest.spyOn(console, 'log');
    greeter.greet();

    expect(spyLog).toHaveBeenCalledWith('Hello, World!');
  });
});
