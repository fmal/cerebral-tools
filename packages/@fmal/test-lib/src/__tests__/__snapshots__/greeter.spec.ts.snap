// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter greets 1`] = `"Hello, World!"`;
